### PR TITLE
chore: Add new issues to triage project

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Triage project update

#### What changes did you make? (Give an overview)

This action ensures that all new issues get added to the Triage project. This replaces the old functionality that was in the GitHub bot (see https://github.com/eslint/eslint-github-bot/pull/184).

Assuming this is correct, we should copy this same file to all the public repos.

#### Is there anything you'd like reviewers to focus on?

Nothing in particular.

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]

File: .github/workflows/add-to-triage.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  issues:
+    types:
+      - opened
+
+jobs:
+  add-to-project:
+    name: Add issue to project
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/add-to-project@RELEASE_VERSION
```
Comment (kecrily) [inline .github/workflows/add-to-triage.yml:13]:
```suggestion
      - uses: actions/add-to-project@v0.4
```

**Review by kecrily — CHANGES_REQUESTED** [review]

File: .github/workflows/add-to-triage.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        with:
+          project-url: https://github.com/orgs/eslint/projects/3
+          github-token: ${{ secrets.PROJECT_BOT_TOKEN }}
```
Comment (mdjermanovic) [inline .github/workflows/add-to-triage.yml:16]:
```suggestion
          github-token: ${{ secrets.PROJECT_BOT_TOKEN }}
          labeled: "triage:no"
          label-operator: NOT
```

To exclude issues with the `triage:no` label. For example, the scheduled release issues.

File: .github/workflows/add-to-triage.yml (hunk 1 above)
Comment (mdjermanovic) [reply .github/workflows/add-to-triage.yml:13]: I just tried this in a test repo and got `` Error: Unable to resolve action `actions/add-to-project@v0.4`, unable to find version `v0.4` ``. Not sure what's wrong with `v0.4`, but `v0.4.0` worked well:

```suggestion
      - uses: actions/add-to-project@v0.4.0
```

**Comment by mdjermanovic** [comment]
The Triage project is private. Is that intentional?

File: .github/workflows/add-to-triage.yml (hunk 2 above)
Comment (nzakas) [reply .github/workflows/add-to-triage.yml:16]: Good call!

**Comment by nzakas** [comment]
The Triage project was always private, I believe.

**Comment by mdjermanovic** [comment]
The old (now closed) Triage project was public, perhaps that was unintentional.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

I'm not merging this now because the project has reached the maximum number of items (1200) so we should either archive or delete the "Completed" items first.

https://github.com/community/community/discussions/9678

**Review by kecrily — APPROVED** [review]
LGTM

**Comment by mdjermanovic** [comment]
I've archived ~1000 completed items, so this should work now.

**Pull request merged**